### DISCUSSION TITLE
Search Field

### DISCUSSION THREAD

**Discussion by Porlin** [twilio-labs/paste#2534] [General] [answered]
Hi There,

Does Paste have a search field with a clear button (maybe a cross icon)? Should we consider using that instead here in this example? If this might be a breaking change we would rule it out.

Flex Teams View example
<img width="535" alt="Screenshot 2022-07-07 at 11 10 46" src="https://user-images.githubusercontent.com/49287701/178957833-3721b8b1-11d4-4950-82b3-a538ba4883c5.png">

I've seen some different usages in Console and on the Paste website eg. search field with a secondary 'Clear filters' button in the console. And on the Paste website in the search feature you use a cross icon. Can you advise which one we can/should use?

**Comment by kelseycavitt** [comment, marked as answer]
Hi @Porlin 👋 !

> _Does Paste have a search field with a clear button (maybe a cross icon)? Should we consider using that instead here in this example? If this might be a breaking change we would rule it out._

For advising what to use, it would be helpful for us to see what you’re wanting to implement at the entire page level, could you share a wider view screenshot or a Figma link? 

If you’re implementing a combobox here, we are actually in the midst of finishing the design of Combobox that does include the ability to display a “clear” button. In [our current release](https://paste.twilio.design/roadmap#2022-07-26), we’re planning to finish the final design of this. In the next release, we [plan to release](https://paste.twilio.design/roadmap#2022-09-06-candidate) the component in Paste.

If you’re referring to [filter group](https://paste.twilio.design/patterns/filter-group/) pattern, there is not a “clear” button within the search input like you have in your screenshot.

For the input component itself, you could implement a clear button in the suffix, as seen below, in Figma (You can manually unhide the layer). We currently don’t support props to have both a prefix and a suffix in Figma out of the box, as it would double the number of variants we’d need to support.

**Reply by kelseycavitt** [reply]
<img width="1792" alt="Screen Shot 2022-07-15 at 1 08 03 PM" src="https://user-images.githubusercontent.com/1484014/179278426-dacf41b2-e6e8-4637-8358-ae65c566633b.png">

**Comment by Porlin** [comment]
Hi @kelseycavitt 

Please see the attached screenshots we have implemented at entire page level.

<img width="1517" alt="Screenshot 2022-07-22 at 09 39 48" src="https://user-images.githubusercontent.com/49287701/180412289-345da857-934c-4917-828e-11de98466de4.png">

<img width="1512" alt="Screenshot 2022-07-22 at 09 39 56" src="https://user-images.githubusercontent.com/49287701/180412319-2e705bf7-0274-4c3d-bc13-6ad94a033253.png">

Thanks for the suggestion to implement a clear button in the suffix, we will try it.
